control-plane/tests/test_service_persistence_restore_readiness.py

```
@@ -379,6 +379,446 @@ def test_service_phase21_backup_restore_and_restore_drill_preserve_record_chain(
+            case_id=promoted_case.case_id,
+            review_owner="analyst-001",
+            intended_outcome="Keep alert-scoped fallback evidence linked to the correct alert.",
+        )
+        action_request = service.create_reviewed_action_request_from_advisory(
+            record_family="recommendation",
+            record_id=recommendation.recommendation_id,
+            requester_identity="analyst-001",
+            recipient_identity="repo-owner-001",
+            message_intent="Record the approved alert-scoped fallback without borrowing other alerts' evidence.",
+            escalation_reason="The approved alert-scoped follow-up cannot wait for the next shift.",
+            expires_at=datetime.now(timezone.utc) + timedelta(hours=4),
```

Ruff (0.15.10)

[error] Undefined name `datetime`

(F821)